homeassistant/components/lutron_caseta/device_trigger.py

```diff
@@ -484,3 +490,15 @@ def get_button_device_by_dr_id(hass: HomeAssistant, device_id: str):
             return device
 
     return None
+
+
+def verify_device_from_domain(hass: HomeAssistant, device_id: str):
+    """Check if HomeAssistant device is port of this domain."""
+    device_registry = dr.async_get(hass)
+    if (device_entry := device_registry.async_get(device_id)) is not None:
+        if device_entry.identifiers is not None:
+            for device_identifiers in device_entry.identifiers:
+                if DOMAIN in device_identifiers:
+                    return True
+
+    return False
```

Comment:
I'd probably write this as 

```python
def is_lutron_caseta_device(hass: HomeAssistant, device_id: str) -> bool:
    """Check if HomeAssistant device is port of this domain."""
    device_registry = dr.async_get(hass)
    if not (device_entry := device_registry.async_get(device_id)):
        return False
    return any(identifier[0] == DOMAIN for identifier in device_entry.identifiers)
```

Reply:
ok, looks good to me.  I'll adjust, test and post a commit

Reply:
tested and working well.  Thanks.